logging/examples_test.go

```diff
@@ -106,6 +106,17 @@ func ExampleHTTPRequest() {
 	lg.Log(httpEntry)
 }
 
+func ExampleToLogEntry() {
+	e := logging.Entry{
+		Payload: "Message",
+	}
+	le, err := logging.ToLogEntry(e, "my-project")
+	if err != nil {
+		// TODO: Handle error.
+	}
+	fmt.Println(le)
```

Comment:
If users still want to write to Cloud Logging (rather than wait for an agents to pick up stdout logs) , how might they do that?

Reply:
My only concern with this: We're adding a function into the [logging module](https://pkg.go.dev/cloud.google.com/go/logging), that can only be used with functions from the [v2 module](https://pkg.go.dev/cloud.google.com/go/logging@v1.3.0/apiv2). And I don't believe this is a common use case for external users. Users who rely on agents/write to stdout, shouldn't use logging libraries in the first place. 

It seems better practice for customers who need this modification, to just directly implement a custom Entry on top of v2 module, rather than modify this higher level module to work with a lower level module. Less overhead. 

@loburm what's the challenge with directly implementing your own Entry/Logger on `v2 module`? 

**We can still merge this PR as a solution, but it would just be for your unique use case.** 

```go
                        // V2 module function                                  // V2 module function
_, err = client.WriteLogEntries(context.Background(), &logpb.WriteLogEntriesRequest{
                                  // V2 module function
		Entries: []*logpb.LogEntry{le},
		LogName: "stdout",
	})
```

Reply:
> ... just directly implement a custom Entry on top of v2 module, rather than modify this higher level module to work with a lower level module

I agree. I think the example code here is a great example.

Reply:
OK, so you basically propose to copy paste Entry and toLogEntry function (and all other helper functions) to v2 module? This is going to create near 150 lines of duplicated code and will increase maintenance cost, because if any issue will be found in one of them, then it would be great to also fix it in another place.

Reply:
I mean: Can you implement Entry and toLogEntry on your own, in your own codebase? Essentially create your own custom local module, which depends on v2 module.

We actually can't edit `v2 module` because it's autogenerated. Our changes to it would be overridden.

Reply:
So a use case for this function is to support customers that want to migrate away from Logger to something more low level (vkit.Client in this case). Implementing it in our own codebase means that we need to copy all the code and monitor for any bug fixes and/or new features.

Another use cases might be for other customers that don't want:

1. Think about how to extract all the data from HttpRequest structure.
2. Think about how to convert text or structured payload to the correctly nested structure inside of LogEntry.

Reply:
I'm not convinced it's a huge use case for external customers. However, I trust that you know best on where to implement logic for agents. Since agents are a big use case for this library, I can LGTM this after tests pass.

Can you change your `ToLogEntry` comment to (1) warn users that this feature is a lower level feature, isn't compatible with `Log/LogSync`, and (2) add a developer comment the fact this feature is necessary to support agents. 

**Parting thoughts:** 
1. Doesn't relying on this library add overhead (latency and size) for your agents instrumentation? Over directly implementing your own custom logic & calling the API endpoints. 
2. In the future, we should refactor this logic at the log/logSync step instead of ToLogEntry, such that users can switch between writing to STDOUT vs Cloud Logging. And at your level, you can just override the logname/fields written to stdout or override log/logsync. 

I instrumented something similar in [Node](https://github.com/googleapis/nodejs-logging/tree/hackathon) for a hackathon. This is better UX, as users wouldn't have to use 2 separate libraries to write logs to Cloud Logging, in addition to stdout.

Reply:
Nicole, thanks for the review. I have updated comments as it was suggested by you.

> 1. Doesn't relying on this library add overhead (latency and size) for your agents instrumentation? Over directly implementing your own custom logic & calling the API endpoints.

Right now we haven't experienced any problems with using Entries. Our agent might get some benefits from creating LogEntry directly, but I don't expect it to be significant gain.